feat(connect): Cardano message signing

jaskp opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

Resolve <!--- link the issue here -->

## Screenshots:

peter-sanderson committed:
chore: dont require default case, be explicit rather then implicit

mroz22 committed:
chore(suite-desktop-core): refactor bridge starting

tomasklim committed:
fix(suite-desktop): restart app on Linux

jaskp committed:
feat(connect): Cardano message signing

mroz22 committed:
feat(transport-bridge): TrezordNode has now the same interface as BridgeProcess

mroz22 committed:
feat(transport-bridge): add binary build

mroz22 committed:
feat(suite-desktop): add --bridge-node flag

mroz22 committed:
feat(transport-bridge): implement stop method

mroz22 committed:
docs(suite-desktop): add a mention about --bridge-node

matejkriz committed:
feat(suite-native): add submitting state for PIN form

komret committed:
chore: updated deprecated jest syntax

komret committed:
chore: update root dependencies

komret committed:
chore: upgrade jest

komret committed:
chore: upgrade jest to 29.7.0

komret committed:
chore: use global jest

komret committed:
chore: upgrade jest-expo

komret committed:
chore: update prettier

komret committed:
chore: use global prettier

komret committed:
chore: remove test scripts for packages without tests

komret committed:
chore: use global rimraf

mroz22 committed:
feat(node-utils): additional features to http server (json parsing, route matching,..)

mroz22 committed:
chore(transport-bridge): migrate from express

marekrjpolak committed:
refactor(blockchain-link): use `createDeferredManager`

mroz22 committed:
chore(suite-dekstop-core): adapt http-receiver to node-utils http updates

komret committed:
chore: update typescript and use global tsc

komret committed:
chore: use global tsx

davidmisiak commented on packages/connect/src/types/api/index.ts:
```diff
@@ -166,6 +167,9 @@ export interface TrezorConnect {
 
     cardanoComposeTransaction: typeof cardanoComposeTransaction;
 
+    // todo: link docs
+    cardanoSignMessage: typeof cardanoSignMessage;
```
I guess the docs are missing?

davidmisiak commented on packages/connect/src/api/cardano/api/cardanoSignMessage.ts:
```diff
@@ -0,0 +1,124 @@
+
+        // TODO: Should we rather allow textual payloads and convert them to hex?
+        // Also, it would be nicer to validate this in the schema by extending typebox
+        // with Type.String({ format: 'hex' }), but I guess that's out of scope.
```
For the sake of simplicity, I would leave this as it is :+1:

davidmisiak replied:
(By this I mean removing the comment :sweat_smile:)

komret committed:
chore: update various dependencies (no major update)

chore: update various dependencies (no major update)

jaskp committed:
fix: Add method to CI and fix protobuf

komret committed:
chore: update sentry dependencies

marekrjpolak committed:
chore(blockchain-link): remove unused error event

komret committed:
chore:update react types and remove them from resolutions

komret committed:
chore(connect-plugin-stellar): update stellar libraries

marekrjpolak committed:
test(connect): global `BlockchainLink` mock

dahaca committed:
fix(suite): fix the notifications page on mobile

davidmisiak commented on packages/connect/e2e/__fixtures__/cardanoSignMessage.ts:
```diff
@@ -0,0 +1,42 @@
+            legacyResults: [legacyResults.beforeMessageSigning],
+        },
+    ],
+};
```
Please don't forget to add more tests. I think a reasonable subset of trezor-firmware test cases would be best.

jaskp committed:
chore(connect): Add doc and more tests

dahaca committed:
fix(suite): fix trade input value not fitting

komret committed:
chore: update react-hook-form and fix types

martykan committed:
chore(connect-web): refactor popupmanager

martykan committed:
chore(connect-web): add channel descriptors to poup, iframe

karliatto committed:
fix(github): browser installation method to issue template

komret committed:
chore(connect-explorer): update markdown-it-replace-link

- I was unable to resolve the type errors.

jaskp committed:
feat(connect-explorer): Add cardanoSignMessage

marekrjpolak committed:
feat(connect): add reconnect functionality to BackendManager

marekrjpolak committed:
test(connect): add BackendManager tests

marekrjpolak committed:
chore(suite): adjust Suite for reconnection in Connect

marekrjpolak committed:
test(suite): adjust tests for reconnection in Connect

dahaca committed:
fix(suite): polish send details

martykan committed:
fix(connect-web): reconnect message channel

STew790 committed:
test(suite-web): eap modal

tomasklim committed:
feat(suite): change language on keyboard shortcut

martykan committed:
fix(connect-web): cross origin fix

komret committed:
chore: update css-loader and remove it from resolutions

peter-sanderson committed:
fix(suite): fix banner for AnalyzeInExplorer

komret committed:
chore: upgrade webpack plugins

mroz22 committed:
chore(connect): improve types, replace any with PassphrasePromptResponse

komret committed:
chore: upgrade framer-motion

komret committed:
chore(suite): bump beta version to 24.3.0

peter-sanderson committed:
chore(suite): make better mapping for colors

matejkriz committed:
chore(suite-native): fix ios build with flipper

- Flipper not allowed by default on ios but breaking the native build

peter-sanderson committed:
chore(suite): even better mapping for elevation to colors

komret committed:
chore(suite): upgrade react-toastify

- Omit closeOnCLick prop as false is the default value now

Nodonisko committed:
chore(ci) GitHub workflow run time analytics (#11066)

* chore(ci): set timeout for CI validation flow

* feat(ci): GH woflows run time analytics

tomasklim committed:
chore(suite): hide polygon trade in trade section

komret committed:
chore(connect-explorer): upgrade markdown-it

tomasklim committed:
feat(suite): isDebugOnly used in getTestnets

komret committed:
chore: upgrade nx

komret committed:
chore: upgrade postcss-loader

peter-sanderson committed:
chore(suite): even more better mapping for elevation

komret committed:
chore: upgrade stylelint

komret committed:
chore: upgrade idb and broadcast-channel

mroz22 committed:
test(suite): adapt database-migrations-test to visual changes

PeKne committed:
fix(suite-native): input top padding (#11056)

- fixes conditonal top padding, the padding should be always the same independent of `isFocused` state

Closes #9472

PeKne committed:
Fix/persisted native message system (#11039)

* fix(suite-native): persisted message system

* fix(suite-native): messages validated on device actions

* fix(suite-native,message-system): fix local fallback on mobile

- Conditional import for some reason start using wrong base path on mobile. Regular import on top of the file fix the issue.
- Potential downside is that local config is fully bundled in the code so it can increase bundle size on web.

---------

Co-authored-by: Matej Kriz <matej@matejkriz.cz>

martykan committed:
fix(connect-popup): fix initTransport in core init

dahaca committed:
feat(theme): update dark theme shadows (#11017)

dahaca committed:
fix(suite): remove a useless animation from the Asset Table (#11016)

AdamSchinzel committed:
fix: empower bugs (#10972)

* fix(suite): spacing of coinjoin state

* fix(suite): coinjoin loading account toggl position

* fix(suite): trade section crypto select inactive

* fix(suite): phishing banner styles

* fix(suite): play store titles

dahaca committed:
feat(suite): fix the discovery button animation (#11022)

peter-sanderson committed:
fix(suite): dropdown to be clickable without collapsing the custom children

peter-sanderson committed:
fix(suite): fix padding for InnerActions

komret committed:
chore(connect): add new CA pubkeys and update timestamp

peter-sanderson committed:
fix(suite): disable action account buttons when error on discovery

jvaclavik committed:
fix(suite): Sidebar button padding fix (#11114)

peter-sanderson committed:
chore(suite): better colors for info/alert cards

komret committed:
chore: minor package updates

tomasklim committed:
fix(suite): fetching fiat rates using eth token definitions

trezor-ci committed:
npm-release: @trezor/blockchain-link 2.1.25

trezor-ci committed:
npm-release: @trezor/blockchain-link-utils 1.0.14

trezor-ci committed:
npm-release: @trezor/blockchain-link-types 1.0.13

trezor-ci committed:
npm-release: @trezor/type-utils 1.0.4

trezor-ci committed:
npm-release: @trezor/connect-analytics 1.0.12

trezor-ci committed:
npm-release: @trezor/analytics 1.0.14

trezor-ci committed:
npm-release: @trezor/connect-common 0.0.28

trezor-ci committed:
npm-release: @trezor/env-utils 1.0.13

trezor-ci committed:
npm-release: @trezor/transport 1.1.24

trezor-ci committed:
npm-release: @trezor/protobuf 1.0.8

trezor-ci committed:
npm-release: @trezor/utxo-lib 2.0.6

trezor-ci committed:
npm-release: @trezor/utils 9.0.21

mroz22 committed:
chore(connect): update changelog

Nodonisko committed:
chore(repo): config cleanups and improvements (TS, Nx...) (#11096)

* chore(repo): config cleanups and improvements (TS, Nx...)

* fix connect e2e

* chore: add eslin cache file to nx cache

* chore: add stylelint cache file to nx

* chore: add ts-node config for to base.tsconfig.json

karliatto committed:
fix(connect-explorer): add fonts to webextension

jvaclavik committed:
fix(suite): Fix TruncateWithTooltip

AdamSchinzel committed:
fix(suite): fetch fiat rates for custom token

peter-sanderson committed:
fix(suite): gradient on elevation

martykan committed:
fix(connect-web): workaround to work with older content-script

jvaclavik committed:
style(suite): Sticky send button (#11136)

AdamSchinzel committed:
fix(suite): fiat input for tokens

karliatto committed:
fix(connect-explorer): docs connect-explorer-webextension

AdamSchinzel committed:
chore(suite): add token copy improvements

jvaclavik committed:
fix(suite): Fix select jumping (#11139)

komret committed:
chore(docker): include updated cypress

jaskp committed:
fix(connect): preferHexDisplay flag and pubKey return

mroz22 committed:
chore(transport-bridge): remove unused deps

tomasklim committed:
chore(suite): use language selector

tomasklim committed:
fix(wallet-utils): token name is optional so do not filter those tokens

martykan committed:
fix(connect-webextension): equalSettings in init, don't focus extension on error

komret committed:
chore: downgrade electron-updater to fix build

karliatto committed:
fix(docker): update playwright image

karliatto committed:
fix(ci): always run connect deploy to dev

Nodonisko committed:
chore(repo): ESLint refactor + speed up 70% (#11143)

* chore: eslint speedup

* chore: disable some react rules for class components

* chore: fix startup command

* chore(deps): dedupe

mroz22 committed:
test(suite-desktop): use --brige-node-test

feat(suite-desktop-core): add --bridge-node-test flag

karliatto committed:
fix(ci): deploy to dev connect production build in release branch

Nodonisko committed:
fix(lint): fix eslint on CI (#11165)

tomasklim committed:
fixup! feat(suite): change language on keyboard shortcut

komret committed:
ci: add commit message check

karliatto committed:
Fix: add missing keys that eslint was complaining about (#11167)

* fix(components): add missings key

* fix(suite): add missing key in iterator

* fix(lint): disable extension rule

---------

Co-authored-by: Carlos Garcia Ortiz karliatto <yo@karliatto.com>
Co-authored-by: Daniel Suchy <suchydan@gmail.com>

martykan committed:
fix(protobuf): codegen devdependencies

Nodonisko committed:
fix(lint): remove unused disabled rules eslint check (#11173)

jvaclavik committed:
style(suite): Update checkbox color (#11166)

jvaclavik committed:
style(suite): Fix onboarding styles (#11168)

* style(suite): Fix onboarding checkbox styles

* style(suite): Update buttons in Final

komret committed:
ci: move fixup check to commit messages check

trezor-ci committed:
chore: crowdin translation update

mroz22 committed:
feat(transport-bridge): support emulator

mroz22 committed:
feat(transport-bridge): add /status endpoint

vytick committed:
feat(suite-native): Select type of added Account (#11119)

mroz22 committed:
chore(protobuf): recreate messages files

mroz22 committed:
chore(protobuf): add eslint --fix to update:protobuf

Nodonisko committed:
chore(lint): enable property shorthand rule (#11180)

Nodonisko committed:
chore: slightly improve formatting scripts (#11177)

komret committed:
chore(components): use package name in import instead of relative path

mroz22 committed:
chore: update trezor-common (51eb23b)

mroz22 committed:
chore(protobuf): add language related messages

mroz22 committed:
chore(protobuf): update with Stellar related messages

komret committed:
ci: create custom fixup commit check

- The action did not work when there was a merge commit.

mroz22 committed:
Revert "fix(connect): rebootToBootloader misbehaviour"

This reverts commit 6b01416d40b4830da0c68ca0e65413c5738126d6.

marekrjpolak committed:
fix(connect): toMessageObject undefined

martykan committed:
fix(suite): version tag color #11106

komret committed:
chore(suite): mark Portuguese as Brazilian

komret committed:
chore(suite): remove beta flag from Japanese

- It is marked as a community language which should be enough.

martykan committed:
fix(suite): #11146 inconsistent border radius in firmware install

juriczech committed:
feat(suite-native): expo eas (#10898)

* feat(suite-native): build binaries for develop

* fix(suite-native): project ID map

* fix(suite-native): version

marekrjpolak committed:
test(connect): ignore __mocks__ in e2e tests

vytick committed:
fix(suite-native): fix dependency version (#11193)

komret committed:
chore(suite): make year dynamic in search tips

martykan committed:
ci: connect-popup-webextension set connect src

mroz22 committed:
chore(transport-bridge): use yarn:g for rimraf

mroz22 committed:
ci: add test protobuf scripts nightly

mroz22 committed:
chore(suite): remove unused translation

mroz22 committed:
chore(connect): improve types of fetch assets util

martykan committed:
fix(connect-web): race condition POPUP.HANDSHAKE before IFRAME.CALL

komret committed:
fix(connect): set min version for rebootToBootloader to correct value

karliatto committed:
ci(github): connect-explorer dev to aws without VPN

AdamSchinzel committed:
chore(suite): replace `selectCoinsLegacy` by fiat rates selectors

PeKne committed:
feat(suite-native): report token contract to analytics (#11206)

Closes #8476

mroz22 committed:
chore(trezor-user-env-link): no implicit bridge start

overcat committed:
feat(connect): add support for StellarClaimClaimableBalanceOp.

overcat committed:
chore: update @stellar/stellar-sdk to v11.2.2

juriczech committed:
fix(suite-native): shorten commit hash (#11203)

juriczech committed:
fix(suite-native): syntax on upload script (#11212)

juriczech committed:
fix(suite-native): remove commit message (#11214)

matejkriz committed:
fix(suite-native): fix missing animation by defining width

matejkriz committed:
chore(suite-native): ignore useless warning in LogBox

karliatto committed:
chore(github): use nvmrc for node version

marekrjpolak committed:
fix(suite-desktop): use `process.env.WS_NO_BUFFER_UTIL` for `ws` lib

karliatto committed:
wip - delete it!

martykan committed:
fix(suite): #10877 timerange dropdown border

peter-sanderson committed:
chore(suite): add new-line before return

karliatto committed:
chore(github): actions to use latest version

We are getting many deprecation warnings in actions since GitHub
is going to use node 20 as
https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

komret committed:
chore(suite): remove analytics for homescreen upload

komret committed:
chore(suite): remove analytics for homescreen choice

jvaclavik committed:
fix(suite): modal for passphrase duplication (#11236)

juriczech committed:
feat(suite-native): production workflow (#11197)

mroz22 committed:
feat(connect): add changeLanguage method

mroz22 committed:
feat(connect-explorer): add changeLanguage method

mroz22 committed:
feat(connect): device.availableTranslations field

mroz22 committed:
chore(suite-common): add device.availableTranslations where needed

mroz22 committed:
chore(connect-explorer): remove unused AsyncSelect

mroz22 committed:
fixup! feat(connect): add changeLanguage method

mroz22 committed:
fixup! fixup! feat(connect): add changeLanguage method

Nodonisko committed:
chore(utils): remove build step requirement from @trezor/utils (#11176)

* chore: remove build step requirement from utils

* chore: dissalow imports from utils/lib

* chore: remove outdated comment from TypedEmitter

* chore: enable tree shaking for connect-iframe build

* chore: bring back usage example comment

mroz22 committed:
feat(connect): rebootToBootloader, add params (boot_command, firmware_header, language_data_length)

jvaclavik committed:
chore(suite): Add stylelint to @trezor/components (#11245)

* chore(suite): Add stylelint to @trezor/components

* fix(suite): Fix stylelint errors

jvaclavik committed:
style(suite): Fix select style (#11244)

peter-sanderson committed:
chore(suite): autofix newlines

peter-sanderson committed:
fix(suite): allow for N screens to be shown on suite (not just hardcoded 2)

peter-sanderson committed:
fix(suite): implement better screen-display for xpub/address confimration

jvaclavik committed:
style(suite): Add Divider component (#11252)

karliatto committed:
fixup! ci(github): connect-explorer dev to aws without VPN

karliatto committed:
fixup! ci(github): connect-explorer dev to aws without VPN

komret committed:
chore(ci): fix commit message check script for CI

AdamSchinzel committed:
fixup! chore(suite): replace `selectCoinsLegacy` by fiat rates selectors

AdamSchinzel committed:
fixup! fixup! chore(suite): replace `selectCoinsLegacy` by fiat rates selectors

AdamSchinzel committed:
fix(suite): save bitcoin calculations

AdamSchinzel committed:
chore(suite): refactor local currency selectors (#11258)

peter-sanderson committed:
fix(suite): separate lines/pages parse login and add test for it (Model One)

juriczech committed:
fix(suite-native): eas develop (#11263)

jvaclavik committed:
style(suite): Chain icon fix (#11248)

karliatto committed:
feat(connect): add CHANNEL_HANDSHAKE_CONFIRM to webextension events

karliatto committed:
test(connect-explorer): add data-test and handshake check to settings screen

karliatto committed:
test(connect-popup): integrate testing for connect-webextension

karliatto committed:
fix(components): move data-test from wrapper to input

karliatto committed:
chore: add build-webextension to tsconfig base ignore

karliatto committed:
feat(connect-webextension): emit events for WEBEXTENSION.CHANNEL_HANDSHAKE_CONFIRM

karliatto committed:
chore(connect-explorer-webextension): delete package - now in connect-explorer/src-webextension

karliatto committed:
chore(docker): add TREZOR_CONNECT_SRC variable

karliatto committed:
chore(ci): adjust connect-explorer-webextension

peter-sanderson committed:
fix(suite): implement remaining trezor models and separate chunk logic

vytick committed:
fix(suite): linting (#11267)

vytick committed:
feat(suite-native): Add coin account via receive (#11209)

Nodonisko committed:
fix(mobile): JS build (#11273)

juriczech committed:
chore(suite): generate icons (#11277)

jvaclavik committed:
style(suite): Fix transaction list paddings (#11274)

karliatto committed:
fix(connect): removed unnecesary logs

juriczech committed:
chore(suite-native): blacklist polygon (#11287)

karliatto committed:
docs(connect-webextension): add examples to readme

peter-sanderson committed:
fix(suite): better code, fix arrows and line-breaking

mroz22 committed:
npm-release: connect-webextension 9.1.12

jvaclavik committed:
fix(suite): discovery buttons tooltip (#11283)

mroz22 committed:
ci: commit-message-check allow release and npm-release

Nodonisko committed:
chore(repo): remove build:lib for some simple packages (#11276)

Nodonisko committed:
chore(repo): remove build from protobuf (#11288)

karliatto committed:
test(connect-popup): move log-page and popup-error-page to new popup-pages

karliatto committed:
test(ci): add new popup-pages test

peter-sanderson committed:
fix(suite): add fix for 3 and more pages

tomasklim committed:
fix(suite): fiat rates are no longed stucked in loading state

tomasklim committed:
fix(suite): download token-definition on selected account update

tomasklim committed:
fix(suite): refetch token fiat rates on successful token def fetch

jvaclavik committed:
suite(feat): 9881 Layout header (#11232)

* feat(suite): Layout header

* fix(suite): fix e2e test

---------

Co-authored-by: dahaca <dahaca17@gmail.com>

AdamSchinzel committed:
fix(suite): `fromWei` and `toWei`check errors (#11266)

jvaclavik committed:
fix(suite): Fix border elevation colors (#11305)

jvaclavik committed:
fix(suite): Fix db migration test (#11324)

komret committed:
test(suite-web): fix database migration test

vytick committed:
chore(suite-native): analytics for add new account (#11325)

peter-sanderson committed:
fix(test): fix e2e test for cardano passphrase

juriczech committed:
chore(suite-native): testnet cannot be imported as bitcoin account (#11329)

* chore(suite-native): validate invalid xpub

* fixup! chore(suite-native): validate invalid xpub

PeKne committed:
feat(suite-native): token definitions redux logic (#11318)

Closes #8434

vytick committed:
feat(suite-native): Add coin behind feature flag (#11332)

jvaclavik committed:
refactor(suite): Refactor UI variant, sizes and alignments (#11280)

juriczech committed:
chore(suite-native): bump app version (#11339)

vytick committed:
feat(suite-native): Add coin account UI polish (#11282)
